staging/src/k8s.io/component-base/metrics/metric.go

```diff
@@ -210,11 +210,6 @@ func (c *selfCollector) Collect(ch chan<- prometheus.Metric) {
 	ch <- c.metric
 }
 
-// metricWithExemplar is an interface that knows how to attach an exemplar to certain supported metric types.
-type metricWithExemplar interface {
-	withExemplar(v float64)
-}
-
```

Comment:
I suppose we'll preferably want to maintain an indication (`var _ satisfiesExemplarPreconditions = &Counter/Histogram{}`) for all types that operate on exemplars, but we can definitely do that later on (in the other PR).